### ISSUE TITLE
Error doing job: Exception in callback _SelectorDatagramTransport._read_ready()

### ISSUE THREAD

**Issue by Jones1982** [rospogrigio/localtuya#887]
## The problem

Greetings,
from the HA event log, it keeps giving me a huge amount (13000 and above) of this reported error.
The log shows the following:

```
Logger: homeassistant
Source: custom_components/localtuya/common.py:117
Integration: LocalTuya integration (documentation, issues)
First occurred: 16:55:26 (13525 occurrences)
Last logged: 23:11:12

Error doing job: Exception in callback _SelectorDatagramTransport._read_ready()
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 1026, in _read_ready
    self._protocol.datagram_received(data, addr)
  File "/config/custom_components/localtuya/discovery.py", line 70, in datagram_received
    self.device_found(decoded)
  File "/config/custom_components/localtuya/discovery.py", line 79, in device_found
    self._callback(device)
  File "/config/custom_components/localtuya/__init__.py", line 105, in _device_discovered
    entry = async_config_entry_by_device_id(hass, device_id)
  File "/config/custom_components/localtuya/common.py", line 117, in async_config_entry_by_device_id
    if device_id in entry.data[CONF_DEVICES]:
KeyError: 'devices'
```

## Environment

- Localtuya version: 4.0.0 
- Last working localtuya version (if known and relevant): 
- Home Assistant Core version: 2022.6.6
- [] Are you using the Home Assistant Tuya Cloud component ? yes
- [] Are you using the Tuya App in parallel ? yes

## Steps to reproduce

1. the various devices connected to localtuya all work, but these errors repeated indefinitely are reported in the HA log.
2. all devices are connected and online on the SmartLife app, and do not give any command problems
3.

## Configuration `configuration.yaml` or `config_flow`

no configuration currently active in configuration.yaml, as replaced by localtuya integration

## Provide Home Assistant taceback/logs

```txt

```

**Comment by homecb** [comment]
This is duplicate of https://github.com/rospogrigio/localtuya/issues/869.

**Comment by findel** [comment]
Looks to me like this is a duplicate of #869

So are #1095 and #930

I suggest everyone else who is experiencing this error also uses #869
